test/adapter/json_api/type_test.rb

```diff
@@ -0,0 +1,61 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
+    module Adapter
+      class JsonApi
+        class TypeTest < ActiveSupport::TestCase
+          class StringTypeSerializer < ActiveModel::Serializer
+            attributes :name
+            type 'profile'
+          end
+
+          class SymbolTypeSerializer < ActiveModel::Serializer
+            attributes :name
+            type :profile
+          end
+
+          def setup
+            @author = Author.new(id: 1, name: 'Steve K.')
+          end
```

Comment:
better to use `setup do @author = Author.new(id: 1, name: 'Steve K.') end` to avoid the need to call `super` (which is lacking here)